pkg/build/controller/strategy/docker.go

```diff
@@ -102,6 +107,15 @@ func (bs *DockerBuildStrategy) CreateBuildPod(build *buildv1.Build, additionalCA
 						EmptyDir: &v1.EmptyDirVolumeSource{},
 					},
 				},
+				{
+					Name: "node-pullsecrets",
+					VolumeSource: v1.VolumeSource{
+						HostPath: &v1.HostPathVolumeSource{
```

Comment:
So this comment here @ricardomaraschini is as much of a broader comment for builds in general as it is for this piece (though we *could* start addressing things with your changes).

Namely, we've added another thing that necessitates a privileged pod.

So, thoughts that come to mind @ricardomaraschini @adambkaplan @bparees :
- we have reqs on the books for devising a build solution that does not require privileged pods, no?
- if I'm wrong there, ignore the rest :-)
- are we hoping to address that with build v1, or will we defer to build v2 for this?
- if not build v1, ignore the rest :-)
- do we have the concise list of places we need privileged?  and if so, should this feature be added to that list?
- and finally, to facilitate that future work, should we start abstracting out privileged requiring elements like this into utility methods, decorating/mutating abstractions, that in the future could consume a config setting and if these privileged elements should be added to the build pod?

thanks

Reply:
> we have reqs on the books for devising a build solution that does not require privileged pods, no?

Unprivileged builds won't happen until we drop support for RHEL7 workers. We need fuse-overlay to give us some level of performance parity.

> are we hoping to address that with build v1, or will we defer to build v2 for this?

It's unclear at this point if builds v2 will be the place where we land the unprivileged builds feature. I like this idea since it lowers the barrier for the upstream community to consume. I.e. builds v2 could be a regular workload on any k8s cluster, _but_ it works best if you have Fedora CoreOS/RHCOS as your host nodes.

> do we have the concise list of places we need privileged? and if so, should this feature be added to that list?

We're already going down the path of adding more host path dependencies with the build image cache enhancement, so I wouldn't worry about this right now.

> and finally, to facilitate that future work, should we start abstracting out privileged requiring elements like this into utility methods, decorating/mutating abstractions, that in the future could consume a config setting and if these privileged elements should be added to the build pod?

I'm not inclined to do this right now. In many respects I'd prefer builds v2 avoid host dependencies so that it could run on any k8s cluster. If there's demand for it, then we can look to refactor things out.

Reply:
> > we have reqs on the books for devising a build solution that does not require privileged pods, no?
> 
> Unprivileged builds won't happen until we drop support for RHEL7 workers. We need fuse-overlay to give us some level of performance parity.
> 
> > are we hoping to address that with build v1, or will we defer to build v2 for this?
> 
> It's unclear at this point if builds v2 will be the place where we land the unprivileged builds feature. I like this idea since it lowers the barrier for the upstream community to consume. I.e. builds v2 could be a regular workload on any k8s cluster, _but_ it works best if you have Fedora CoreOS/RHCOS as your host nodes.
> 
> > do we have the concise list of places we need privileged? and if so, should this feature be added to that list?
> 
> We're already going down the path of adding more host path dependencies with the build image cache enhancement, so I wouldn't worry about this right now.
> 
> > and finally, to facilitate that future work, should we start abstracting out privileged requiring elements like this into utility methods, decorating/mutating abstractions, that in the future could consume a config setting and if these privileged elements should be added to the build pod?
> 
> I'm not inclined to do this right now. In many respects I'd prefer builds v2 avoid host dependencies so that it could run on any k8s cluster. If there's demand for it, then we can look to refactor things out.

my last comment was meant to be in the context of build v1, not build v2.  Regardless, it only has bearing if the RHEL7/fuse_overlay items you noted initially are addressed.